brut.apktool/apktool-lib/src/main/java/brut/androlib/res/AndrolibResources.java

```diff
@@ -319,41 +313,162 @@ public String checkTargetSdkVersionBounds() {
+
+            try {
+                tempResourcesZip = File.createTempFile("BRUT", ".zip");
+                tempResourcesZip.deleteOnExit();
+
+                cmd.add("-o");
+                cmd.add(tempResourcesZip.getAbsolutePath());
 
                 if (apkOptions.verbose) {
-                    LOGGER.info(aaptFile.getPath() + " being used as aapt location.");
+                    cmd.add("-v");
                 }
-            } else {
-                LOGGER.warning("aapt location could not be found. Defaulting back to default");
 
                 try {
-                    cmd.add(getAaptBinaryFile().getAbsolutePath());
-                } catch (BrutException ignored) {
-                    cmd.add("aapt");
+                    OS.exec(cmd.toArray(new String[0]));
+                    if (apkOptions.verbose) {
+                        LOGGER.info("aapt2 compile command ran: ");
+                        LOGGER.info(cmd.toString());
+                    }
+                } catch (BrutException ex) {
+                    throw new AndrolibException(ex);
                 }
+            } catch (IOException ex) {
+                throw new AndrolibException(ex);
             }
-        } else {
-            try {
-                cmd.add(getAaptBinaryFile().getAbsolutePath());
-            } catch (BrutException ignored) {
-                cmd.add("aapt");
+        }
+
+        if (manifest == null) {
+            return;
+        }
+
+        // Link them into the final apk, reusing our old command after clearing for the aapt2 binary
+        cmd = new ArrayList<>(compileCommand);
+        cmd.add("link");
+
+        cmd.add("-o");
+        cmd.add(apkFile.getAbsolutePath());
+
+        if (mPackageId != null && ! mSharedLibrary) {
+            cmd.add("--package-id");
+            cmd.add(mPackageId);
+        }
+
+        if (mSharedLibrary) {
+            cmd.add("--shared-lib");
+        }
+
+        if (mMinSdkVersion != null) {
+            cmd.add("--min-sdk-version");
+            cmd.add(mMinSdkVersion);
+        }
+
+        if (mTargetSdkVersion != null) {
+            cmd.add("--target-sdk-version");
+            cmd.add(checkTargetSdkVersionBounds());
+        }
+
+        if (mPackageRenamed != null) {
+            cmd.add("--rename-manifest-package");
+            cmd.add(mPackageRenamed);
+
+            cmd.add("--rename-instrumentation-target-package");
+            cmd.add(mPackageRenamed);
+        }
+
+        if (mVersionCode != null) {
+            cmd.add("--version-code");
+            cmd.add(mVersionCode);
+        }
+
+        if (mVersionName != null) {
+            cmd.add("--version-name");
+            cmd.add(mVersionName);
+        }
+
+        // Disable automatic changes
+        cmd.add("--no-auto-version");
+        cmd.add("--no-version-vectors");
+        cmd.add("--no-version-transitions");
+        cmd.add("--no-resource-deduping");
+
+        // TODO Sparse Resources
```

Comment:
We have the sparse-resource flag in our `apktool.yml` file but haven't read it back into the system.